### PR TITLE
be able to navigate backward in the manual

### PR DESCRIPTION
**Pull request by introkun**

### CODE REVIEW

File: src/infoPanel/infoPanel.c [hunk 1]
```diff
@@ -88,10 +148,56 @@ int main(int argc, char *argv[])
...
+				{
+					if (navigating_forward)
+					{
+						image_index++;
+					}
+					else
+					{
+						image_index--;
+					}
+					drawImage(images_paths[image_index], screen);
```
Comment (Aemiii91) [inline src/infoPanel/infoPanel.c:185]: It seems quite inefficient to load the images every time they're drawn? It's the navigation slowed by this. You just need to store the surface pointers in an array, and remember to free each one at exit.

**Review by Aemiii91 — CHANGES_REQUESTED** [review]
Please improve the image loading

File: src/infoPanel/infoPanel.c (hunk 1 above)
Comment (introkun) [reply src/infoPanel/infoPanel.c]: fixed

**Comment by Aemiii91** [comment]
If I understand the code somewhat, you're still loading an image every time navigation happens? I don't think this is any more efficient than your other implementation?

The reason for using a cache was to make navigation faster, there's no reason the cache couldn't be 100 images, then you could (if you ever hit 100) reset the cache's index to 0, thereby just overwriting the first images loaded.

Also, did you test this on-device? The miyoo mini is much slower at loading images than on desktop, so it's very noticeable when on-device - that said, I haven't tested it on-device either, so it could be fine.

**Comment by introkun** [comment]
@Aemiii91 I've implemented a cache for 3 images (current slide, previous and next one). When user navigates to the next slide it will preload next image to be ready to render it at the next step. There's no way to navigate to a specific page, so the solution is quite good. It holds memory for 3 images only. If user will return back and forth the app will not touch SDCARD.

**Comment by introkun** [comment]
Yes, I've tested on device and provided video to the channel ;)

**Comment by introkun** [comment]
When navigating to the 1st image it loads current slide and 2nd slide.
When navigating to the 2nd slide it renders image from the memory and preloads 3rd slide. etc.

**Comment by Aemiii91** [comment]
And on 3rd slide it loads 4th image, 4th slide it loads 5th image, and so on... So every time the user navigates forward there's a blocking IMG_Load call, right?

**Comment by introkun** [comment]
> And on 3rd slide it loads 4th image, 4th slide it loads 5th image, and so on... So every time the user navigates forward there's a blocking IMG_Load call, right?

Yes. Not sure how this affects performance on the device. As you can see on the video there's no lags.

**Review by Aemiii91 — APPROVED** [review]

**Pull request merged**